.travis.yml

```diff
@@ -16,9 +16,26 @@ addons:
       - cmake
       - ninja-build
 
+cache:
+  directories:
+    - $HOME/downloads
+    - $HOME/toolchain
+
 install:
-  - 'wget -O ~/toolchain.sh "${TOOLCHAINS_URL}/${TOOLCHAIN_NAME}"'
-  - chmod +x ~/toolchain.sh && ~/toolchain.sh -d ~/toolchain -y
+  - |-
+    mkdir -p ~/downloads
+    wget --tries=3 -N -O ~/downloads/toolchain.sh "${TOOLCHAINS_URL}/${TOOLCHAIN_NAME}"
```

Comment:
This doesn't work, I just realized that `wget` prints

```
WARNING: timestamping does nothing in combination with -O. See the manual
```

Using `curl --time-cond` should work, so I'll update the PR to use that instead.

Reply:
This works just perfect. I tried triggering rebuilds a few times with the cache empty, and with data in the cache. A build using a cached toolchain is around 25s faster than a build that downloads and installs it.